packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+  });
+
+  /// The style to apply to this span.
+  ///
+  /// The [style] is also applied to any child spans when this is an instance
+  /// of [TextSpan].
+  final TextStyle style;
+
+  /// A gesture recognizer that will receive events that hit this span.
+  ///
+  /// [InlineSpan] itself does not implement hit testing or event dispatch. The
+  /// object that manages the [InlineSpan] painting is also responsible for
+  /// dispatching events. In the rendering library, that is the
+  /// [RenderParagraph] object, which corresponds to the [RichText] widget in
+  /// the widgets layer; these objects do not bubble events in [InlineSpan]s, so a
+  /// [recognizer] is only effective for events that directly hit the [text] of
+  /// that [InlineSpan], not any of its [children].
+  ///
+  /// [InlineSpan] also does not manage the lifetime of the gesture recognizer.
+  /// The code that owns the [GestureRecognizer] object must call
+  /// [GestureRecognizer.dispose] when the [InlineSpan] object is no longer used.
+  ///
+  /// {@tool sample}
+  ///
+  /// This example shows how to manage the lifetime of a gesture recognizer
+  /// provided to a [InlineSpan] object. It defines a `BuzzingText` widget which
+  /// uses the [HapticFeedback] class to vibrate the device when the user
+  /// long-presses the "find the" span, which is underlined in wavy green. The
+  /// hit-testing is handled by the [RichText] widget.
+  ///
+  /// ```dart
+  /// class BuzzingText extends StatefulWidget {
+  ///   @override
+  ///   _BuzzingTextState createState() => _BuzzingTextState();
+  /// }
+  ///
+  /// class _BuzzingTextState extends State<BuzzingText> {
+  ///   LongPressGestureRecognizer _longPressRecognizer;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     _longPressRecognizer = LongPressGestureRecognizer()
+  ///       ..onLongPress = _handlePress;
+  ///   }
+  ///
+  ///   @override
+  ///   void dispose() {
+  ///     _longPressRecognizer.dispose();
+  ///     super.dispose();
+  ///   }
+  ///
+  ///   void _handlePress() {
+  ///     HapticFeedback.vibrate();
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return RichText(
+  ///       text: TextSpan(
+  ///         text: 'Can you ',
+  ///         style: TextStyle(color: Colors.black),
+  ///         children: <InlineSpan>[
+  ///           TextSpan(
+  ///             text: 'find the',
+  ///             style: TextStyle(
+  ///               color: Colors.green,
+  ///               decoration: TextDecoration.underline,
+  ///               decorationStyle: TextDecorationStyle.wavy,
+  ///             ),
+  ///             recognizer: _longPressRecognizer,
+  ///           ),
+  ///           TextSpan(
+  ///             text: ' secret?',
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  /// {@end-tool}
+  final GestureRecognizer recognizer;
```

Comment:
I think this should be on TextSpan (if anywhere). It's not a great API in the first place, but for PlaceholderSpan it's particularly bad since people could just put actual GestureDetectors in their widget subtree.